app/controllers/mixins/ems_common_angular.rb

```diff
@@ -128,7 +128,8 @@ def get_task_args(ems)
         connect_opts = [MiqPassword.encrypt(params[:amqp_password]), params.to_hash.symbolize_keys.slice(*OPENSTACK_AMQP_PARAMS)] if params[:cred_type] == "amqp"
         connect_opts
       when 'ManageIQ::Providers::Amazon::CloudManager'
-        [user, password, :EC2, params[:provider_region], ems.http_proxy_uri, true]
+        uri = URI.parse(params[:default_url])
```

Comment:
incoming params should be escaped for safety
http://devdocs.io/ruby~2.3/uri#method-c-parse:
`It's recommended to first ::escape the provided uri_str if there are any invalid URI characters.`
http://devdocs.io/ruby~2.3/uri/escape#method-i-escape

Reply:
Related comment here: https://github.com/ManageIQ/manageiq-providers-amazon/pull/444#issuecomment-391641011 (need to decide where escaping will be made)